.script/tests/detectionTemplateSchemaValidation/DetectionTemplateSchemaValidationTests.cs

```diff
@@ -10,19 +12,50 @@
 
 namespace Kqlvalidations.Tests
 {
-    public class DetectionTemplateStructureValidationTests
+    public class DetectionTemplateSchemaValidationTests
     {
         private static readonly string DetectionPath = DetectionsYamlFilesTestData.GetDetectionPath();
 
         [Theory]
         [ClassData(typeof(DetectionsYamlFilesTestData))]
         public void Validate_DetectionTemplates_HaveValidTemplateStructure(string detectionsYamlFileName)
+        {
+            Action<ScheduledTemplateInternalModel> validateConnectorSchema = (ScheduledTemplateInternalModel templateObject) => {
+                var validationContext = new ValidationContext(templateObject);
+                Validator.ValidateObject(templateObject, validationContext, true);
+            };
+
+            ValidateDetectionTemplate(detectionsYamlFileName, validateConnectorSchema);
+        }
+
+        [Theory]
+        [ClassData(typeof(DetectionsYamlFilesTestData))]
+        public void Validate_DetectionTemplates_HaveValidConnectorsIds(string detectionsYamlFileName)
+        {
+            Action<ScheduledTemplateInternalModel> validateConnectorsIds = (ScheduledTemplateInternalModel templateObject) => {
+                List<string> connectorIds = templateObject.RequiredDataConnectors
+                    .Where(requiredDataConnector => requiredDataConnector.ConnectorId != null)
```

Comment:
are there any templates with connector id = null?

Reply:
no, but could be as it's not required ( rembmer I saw that)